src/storage/leveldb/token_index.ts

```diff
@@ -254,9 +254,15 @@ export default class LevelTokenIndex implements IKVTokenIndex {
 
   /**
    * Clear all entries from the database.
+   * @param {boolean} skipRegistered skip deleting the registered tokens
    * @returns {Promise<void>}
    */
-  async clear(): Promise<void> {
-    await this.tokenDB.db.clear();
+  async clear(skipRegistered: boolean = false): Promise<void> {
+    if (skipRegistered) {
```

Comment:
I believe this check should contain a negative, right?
```suggestion
    if (!skipRegistered) {
```

Reply:
It's correct, `this.tokenDB.db` refers to the database containing all token databases or sublevels (token, metadata and registered tokens).
If we want to skip the registered tokens sublevel we have to manually clear the token and metadata ones.